activerecord/lib/active_record/encryption/scheme.rb

```diff
@@ -87,8 +84,15 @@ def validate_credential(key, error_message = "is not configured")
           end
         end
 
+        def key_provider_from_config
+          ActiveRecord::Encryption.context.key_provider
+        end
+
         def build_key_provider
+          validate_keys!
+
           return DerivedSecretKeyProvider.new(@key) if @key.present?
+          return DerivedSecretKeyProvider.new(ActiveRecord::Encryption.config.primary_key) if ActiveRecord::Encryption.config.primary_key.present?
```

Comment:
We want to build the `DerivedSecretKeyProvider` using `@key` to make the [`key:` configuration option work](https://edgeguides.rubyonrails.org/active_record_encryption.html#model-specific-keys).

Reply:
Yep - that's what's happening on line 94, right?

Reply:
Oh you're right!